src/families/tron/react.js

```diff
@@ -68,3 +88,165 @@ export const formatVotes = (
       }))
     : [];
 };
+
+/** Search filters for SR list */
+const searchFilter = (query?: string) => ({
+  name,
+  address
+}: {
+  name: ?string,
+  address: string
+}) => {
+  if (!query) return true;
+  const terms = `${name || ""} ${address}`;
+  return terms.toLowerCase().includes(query.toLowerCase().trim());
+};
+
+/** Hook to search and sort SR list according to initial votes and query */
+export function useSortedSr(
+  search: string,
+  superRepresentatives: SuperRepresentative[],
+  votes: Vote[]
+): {
+  sr: SuperRepresentative,
+  name: ?string,
+  address: string,
+  rank: number,
+  isSR: boolean
+}[] {
+  const [initialVotes] = useState(votes.map(({ address }) => address));
```

Comment:
useMemo?

Reply:
not here i don't want it to recompute just initialize and that it :+1:

Reply:
In that case, i would suggest useRef. useState shouldn't be used without set func.

Reply:
is there a reason why @LukeSugiura ? I used that pattern at a few places too and i didn't find a better one. useRef is not equivalent, is it?